core/settings/qPref.h

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef QPREF_H
+#define QPREF_H
+
+
+#include <QObject>
+#include "core/pref.h"
+
+
+
+class qPref : public QObject {
+	Q_OBJECT
+	Q_ENUMS(cloud_status);
+
+public:
+	qPref(QObject *parent = NULL) : QObject(parent) {};
+	~qPref() {};
+	static qPref *instance();
+
+	// Load/Sync local settings (disk) and struct preference
+	void loadSync(bool doSync);
+
+public:
+	enum cloud_status {
```

Comment:
I admit that I haven't really looked at the other PR, because the preferences code is the part which I am the least fond of.

So just a random note: Why would `cloud_status` belong to the preferences-class? This seems to me fundamentally wrong. So I wonder whether a better approach would be to declutter the preferences first before doing whatever this PR is doing?

Reply:
I agree that this is not a preference - but it is stored with the preferences... now you can argue "why do we do that?" and that's an interesting question. Preferences and cloud status are both part of the persistent state that we maintain across runs. We can now consider renaming the thing that we store - but I'm not sure that adds a lot of value.

Reply:
>  Preferences and cloud status are both part of the persistent state that we maintain across runs.

So are many other things which are *not* stored in the preferences (last used directories, thumbnails, ...). Just grep for `QSettings`.

> but I'm not sure that adds a lot of value

I guess that depends on your priorities. But since this is a major rewrite/renaming anyway, I think we should ask such questions.

Even the code says:
```
//TODO: Change this to the Cloud Storage Stuff, not preferences.
```

BTW: I really dislike the verbosity of the whole preferences thing, so just seeing `+356 −191` makes me sad. I reckon that's a temporary thing?

Reply:
You are correct - this is the time to do things right. So we should move cloud storage out from preferences. That doesn't need to happen in this particular patch series, though.

@tcanabrava had worked on something to create the preferences and all the boilerplate code from much more concise configuration data. I don't know if that code would be an alternative to what Jan is working on.